addons/product/models/product.py

```diff
@@ -229,7 +229,7 @@ def _compute_product_code(self):
     def _compute_partner_ref(self):
         for supplier_info in self.seller_ids:
             if supplier_info.name.id == self._context.get('partner_id'):
-                product_name = supplier_info.product_name or self.default_code or self.name
+                product_name = supplier_info.product_name or supplier_info.product_code  or self.name
```

Comment:
Lap doesn't agree with this change. 

The "Customer Ref" (-> Sales) shouldn't use the vendor internal code (-> Purchase). 

I'm gonna drop this commit.